pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -737,8 +737,9 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 			pods, err := nc.getPodsAssignedToNode(node.Name)
 			if err != nil {
 				utilruntime.HandleError(fmt.Errorf("unable to list pods of node %v: %v", node.Name, err))
-				if currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue {
-					// If error happened during node status transition (Ready -> NotReady)
+				if (currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue) ||
+					(currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status != currentReadyCondition.Status) {
```

Comment:
I think this code can be changed to this：
currentReadyCondition.Status == v1.ConditionUnknown && observedReadyCondition.Status != v1.ConditionUnknown
@yylt

Reply:
the reason for doing this?